### PR TITLE
Policy REST API

### PR DESCRIPTION
**Pull request by joshblakeley**
## Description

This PR creates a way to manage policies in Tyk CE using a REST API instead of the single file approach currently. It makes policies act in very much the same way as API when using CE. A directory is specified in the Tyk config in the new `policies.policy_path` field. Policies are then stored in that folder with their ID as a filename i.e. `default-policy.json`. CRUD operations can be carried out by calling `{GATEWAYHOST}:{PORT}/tyk/policies` endpoint with the policy ID passed in the last part of Path.

## Related Issue

Fixes https://github.com/TykTechnologies/tyk/issues/2379

## Motivation and Context

Means simpler way of managing policies - also unblocks development of Kubernetes Operator for the OSS Gateway.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ ] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ ] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

**Review by gernest — CHANGES_REQUESTED** [review]
Looks good, I left a few comments. After they are addressed I will take another look.

**Comment by joshblakeley** [comment]
Thanks for the review @gernest. Have made suggested changes.

File: gateway/server.go [hunk 1]
```diff
@@ -138,6 +138,13 @@ func setTestMode(v bool) {
 	runningTestsMu.Unlock()
 }
 
+func getPolicy(polID string) user.Policy {
+	apisMu.RLock()
+	pol := policiesByID[polID]
+	apisMu.RUnlock()
```
Comment (gernest) [inline gateway/server.go:144]:
```suggestion
	policiesMu.RLock()
	pol := policiesByID[polID]
	policiesMu.RUnlock()
```

**Review by gernest — CHANGES_REQUESTED** [review]
I left another comment. We also need tests to cover all the cases.

File: gateway/server.go (hunk 1 above)
Comment (joshblakeley) [reply gateway/server.go:144]: I was worried you would say that :D

**Review by buger — APPROVED** [review]
Fix Linter and CI in general and looks solid 👍🏻

File: config/config.go [hunk 2]
```diff
@@ -85,7 +85,8 @@ type PoliciesConfig struct {
 	// If you set this value to `true`, then the id parameter in a stored policy (or imported policy using the Dashboard API), will be used instead of the internal ID.
 	//
 	// This option should only be used when moving an installation to a new database.
-	AllowExplicitPolicyID bool `json:"allow_explicit_policy_id"`
+	AllowExplicitPolicyID bool   `json:"allow_explicit_policy_id"`
+	PolicyPath            string `json:"policy_path"`
```
Comment (buger) [inline config/config.go:89]: Do not forget adding a comment to all fields, since it will be used to auto-generate docs.

**Review by buger — APPROVED** [review]
Add comment for `PolicyPath` option, otherwise looks good!

File: config/config.go (hunk 2 above)
Comment (vaske) [reply config/config.go:89]: Oh didn't know that, adding it

**Review by gernest — APPROVED** [review]

**Comment by vaske** [comment]
merged manually to release-4-lts

**Pull request merged**